core/src/main/java/io/parsingdata/metal/Util.java

```diff
@@ -51,13 +52,7 @@ private Util() {}
 
     public static String tokensToString(final Token[] tokens) {
         checkNotNull(tokens, "tokens");
-        if (tokens.length == 0) { return ""; }
-        final StringBuilder outString = new StringBuilder();
-        for (int i = 0; i < tokens.length - 1; i++) {
-            outString.append(tokens[i].toString());
-            outString.append(", ");
-        }
-        return outString.append(tokens[tokens.length - 1]).toString();
+        return Arrays.stream(tokens).map(t -> t.toString()).reduce((l, r) -> l + ", " + r).orElse("");
```

Comment:
Just wondering, is Java 8 "smart enough" to use a `StringBuilder` internally? I think in certain situations the compiler does optimize string concatenation like this.